build-system/tasks/performance-urls.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const colors = require('ansi-colors');
+const fs = require('fs');
+const gulp = require('gulp');
+const log = require('fancy-log');
+const path = require('path');
+const through2 = require('through2');
+
+const CONFIG_PATH = 'build-system/tasks/performance/config.json';
+const LOCAL_HOST_URL = 'http://localhost:8000/';
+
+/**
+ * Entry point for 'gulp performance-urls'
+ * Check if all localhost urls in performance/config.json exist
+ * @return {!Promise}
+ */
+async function performanceUrls() {
+  return gulp.src([CONFIG_PATH]).pipe(
+    through2.obj(function (file) {
+      let obj;
+      try {
+        obj = JSON.parse(file.contents.toString());
+      } catch (e) {
+        log(colors.yellow(`Could not parse ${CONFIG_PATH}. `));
+        return;
+      }
+      const urls = obj.handlers.flatMap((handler) =>
+        handler.urls
+          .filter((url) => url.startsWith(LOCAL_HOST_URL))
+          .map((url) =>
+            path.join(__dirname, '../../', url.split(LOCAL_HOST_URL)[1])
+          )
+      );
+      for (const url of urls) {
+        if (!fs.existsSync(url)) {
+          log(colors.red(url + ' does not exist.'));
+          process.exitCode = 1;
```

Comment:
Do I need to do anything else here to fail the task?

Reply:
Nope, this is fine!

Reply:
Optional: Now that we're on `gulp` v4, here's an alternative way that has the added benefit of printing a clear message without the mostly useless `nodejs` call stack:

https://github.com/ampproject/amphtml/blob/79b463476ec9b7e588bde89c060433bbac54620b/build-system/tasks/check-sourcemaps.js#L37-L46

Note to self: I wonder if it's worth making this a util method that other tasks can also use.